src/test/java/com/getindata/kafka/connect/iceberg/sink/TestIcebergUtil.java

```diff
@@ -242,6 +244,38 @@ public void coerceDebeziumTemporalTypesEnabledBehavior(@TempDir Path localWareho
         assertEquals(record.getField("ship_time_zoned"), OffsetTime.parse("20:32:46.821144Z").toLocalTime());
     }
 
+    @Test
+    public void listStructSchemaHandling()
+      throws JsonProcessingException {
+        IcebergChangeEvent e = new IcebergChangeEvent("test",
+                                          MAPPER.readTree(debeziumMetadataSchema).get("payload"), null,
+                                          MAPPER.readTree(debeziumMetadataSchema).get("schema"), null,
+                                          defaultConfiguration
+        );
+        Schema schema = e.icebergSchema();
+        String schemaString = schema.toString();
+
+        GenericRecord record = e.asIcebergRecord(schema);
+
+        assertTrue(schemaString.contains("data_collections: optional list<struct"));
+
+        GenericRecord innerRecord = (GenericRecord) ((ArrayList) record.getField("data_collections")).get(0);
+        Object value = innerRecord.getField("data_collection");
+        assertTrue(value.equals("public.mine"));
+
+        value = innerRecord.getField("event_count");
+        assertTrue((long) value == 1);
```

Comment:
It's better to use `assertEquals` because then the test runner can give you helpful output like "expected X but got Y". The order of the arguments is `expected, actual` for this.

Also the casts aren't needed -- you can compare an `Object` to any other object and a type check will be included:


```suggestion
        assertEquals("public.mine", value);

        value = innerRecord.getField("event_count");
        assertEquals(1L, value);
```